### ISSUE TITLE
Module STD : Yet another error with crtdbg header

### ISSUE THREAD

**Issue by Cazayus** [microsoft/STL#4668]
# Describe the bug
It is possible to work around the issue #4666 with a #define, but the following code still fails to compile with another error.
Seemed worth reporting separately just in case.

# Command-line test case
```
C:\Temp>type repro.cpp
import std;
#define align_val_t ignore_align_val_t // only here to silence the error of issue #4666

class SomeClass
{
    std::span<size_t> method1();
    void method2() { method1(); }
};

#include <crtdbg.h>

C:\Temp>cl.exe /std:c++latest /EHsc /nologo /c .\repro.cpp

C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33807\include\vcruntime_new.h(35): error C2011: 'std::nothrow_t': 'struct' type redefinition
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33807\include\vcruntime_new.h(35): note: see declaration of 'std::nothrow_t'
```

# Expected behavior
It should compile but does not

# STL version
```
    Microsoft Visual Studio Community 2022
    Version 17.10.0 Preview 7.0
```

**Comment by StephanTLavavej** [comment]
Also a duplicate of #4666. While the error is different (`std::nothrow_t` instead of `std::align_val_t`), the root cause is the same - `<crtdbg.h>` drags in VCRuntime, and VCRuntime has to export a bit of Standard Library machinery, so this is the import-before-include scenario that's blocked by our implementation bug/limitation.

Again, I appreciate these comprehensive reports, and thanks for being an early adopter of modules. :heart_eyes_cat:
